internal/blockchain/ethereum/ethereum.go

```diff
@@ -255,103 +247,32 @@ func (e *Ethereum) Start() (err error) {
-		address = entry.GetString(FireFlyContractAddress)
-		if address == "" {
-			return "", "", i18n.NewError(ctx, coremsgs.MsgMissingPluginConfig, "address", "blockchain.ethereum.fireflyContract")
-		}
-		fromBlock = entry.GetString(FireFlyContractFromBlock)
-	} else {
-		// Old config (attributes under "ethconnect")
-		address = e.ethconnectConf.GetString(EthconnectConfigInstanceDeprecated)
-		if address != "" {
-			log.L(ctx).Warnf("The %s.%s config key has been deprecated. Please use %s.%s instead",
-				EthconnectConfigKey, EthconnectConfigInstanceDeprecated,
-				FireFlyContractConfigKey, FireFlyContractAddress)
-		} else {
-			return "", "", i18n.NewError(ctx, coremsgs.MsgMissingPluginConfig, "instance", "blockchain.ethereum.ethconnect")
-		}
-		fromBlock = e.ethconnectConf.GetString(EthconnectConfigFromBlockDeprecated)
-		if fromBlock != "" {
-			log.L(ctx).Warnf("The %s.%s config key has been deprecated. Please use %s.%s instead",
-				EthconnectConfigKey, EthconnectConfigFromBlockDeprecated,
-				FireFlyContractConfigKey, FireFlyContractFromBlock)
-		}
-	}
-
-	// Backwards compatibility from when instance path was not a contract address
-	if strings.HasPrefix(strings.ToLower(address), "/contracts/") {
-		address, err = e.getContractAddress(ctx, address)
-		if err != nil {
-			return "", "", err
-		}
-	} else if strings.HasPrefix(address, "/instances/") {
-		address = strings.Replace(address, "/instances/", "", 1)
-	}
-
-	address, err = validateEthAddress(ctx, address)
-	return address, fromBlock, err
+func (e *Ethereum) Capabilities() *blockchain.Capabilities {
+	return e.capabilities
 }
 
-func (e *Ethereum) ConfigureContract(ctx context.Context, contracts *core.FireFlyContracts) (err error) {
-
-	log.L(ctx).Infof("Resolving FireFly contract at index %d", contracts.Active.Index)
-	address, fromBlock, err := e.resolveFireFlyContract(ctx, contracts.Active.Index)
+func (e *Ethereum) AddFireflySubscription(ctx context.Context, namespace string, location *fftypes.JSONAny, firstEvent string) (string, error) {
+	ethLocation, err := parseContractLocation(ctx, location)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	sub, err := e.streams.ensureFireFlySubscription(ctx, address, fromBlock, e.streamID, batchPinEventABI)
-	if err == nil {
-		var version int
-		version, err = e.getNetworkVersion(ctx, address)
-		if err == nil {
-			e.fireflyContract.mux.Lock()
-			e.fireflyContract.address = address
-			e.fireflyContract.fromBlock = fromBlock
-			e.fireflyContract.networkVersion = version
-			e.fireflyContract.subscription = sub.ID
-			e.fireflyContract.mux.Unlock()
-			contracts.Active.Info = fftypes.JSONObject{
-				"address":      address,
-				"fromBlock":    fromBlock,
-				"subscription": sub.ID,
-			}
-		}
+	sub, err := e.streams.ensureFireFlySubscription(ctx, namespace, ethLocation.Address, firstEvent, e.streamID, batchPinEventABI)
+	if err != nil {
+		return "", err
 	}
-	return err
-}
+	e.subs[sub.ID] = namespace
```

Comment:
We will probably need to save the namespace AND network version here. Ultimately there needs to be a logic branch in the event handling, where for "V1" we expect to receive a namespace in every BatchPin event, but for "V2" we infer the namespace based on which subscription ID produced it.

Doesn't require a change here, just a note for whichever of us does the next piece.